SYCL/Regression/depends_on.cpp

```diff
@@ -0,0 +1,65 @@
+//==-- depends_on.cpp.cpp ----------------------------------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// RUN: %clangxx -fsycl -fsycl-unnamed-lambda -fsycl-targets=%sycl_triple %s -o %t.out
+// RUN: %HOST_RUN_PLACEHOLDER %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+// RUN: %ACC_RUN_PLACEHOLDER %t.out
+
+#include <CL/sycl.hpp>
+
+int main() {
+  const int N = 2;
+  cl::sycl::device D1{cl::sycl::default_selector()};
+  cl::sycl::device D2{cl::sycl::default_selector()};
+  cl::sycl::context C1{D1};
+  cl::sycl::context C2{D2};
+  cl::sycl::queue Q1(C1, D1);
+  cl::sycl::queue Q2(C2, D2);
+
+  int *shared_array1a = cl::sycl::malloc_shared<int>(N, Q1);
+  int *shared_array2a = cl::sycl::malloc_shared<int>(N, Q2);
+  int *shared_array1b = cl::sycl::malloc_shared<int>(N, Q1);
+  int *shared_array2b = cl::sycl::malloc_shared<int>(N, Q2);
+
+  for (int i = 0; i < N; i++) {
+    shared_array1a[i] = 0;
+    shared_array2a[i] = 0;
+    shared_array1b[i] = 0;
+    shared_array2b[i] = 0;
+  }
+
+  auto eA = Q1.submit([&](cl::sycl::handler &h) {
+    h.parallel_for(N, [=](cl::sycl::id<1> i) {
+      shared_array1a[i] = i + 110;
+      shared_array1b[i] = i + 120;
+    }); // Task A
+  });
+  eA.wait();
+
+  auto eB = Q2.submit([&](cl::sycl::handler &h) {
+    h.depends_on(eA);
+    h.parallel_for(N, [=](cl::sycl::id<1> i) {
+      shared_array2a[i] = i + 210;
+      shared_array2b[i] = i + 220;
+    }); // Task B
+  });
+  eB.wait();
+
+  assert(shared_array1a[0] == 110);
+  assert(shared_array1b[0] == 120);
+  assert(shared_array1a[1] == 111);
+  assert(shared_array1b[1] == 121);
+
+  assert(shared_array2a[0] == 210);
+  assert(shared_array2b[0] == 220);
+  assert(shared_array2a[1] == 211);
+  assert(shared_array2b[1] == 221);
```

Comment:
> Previously, when the "depends_on" function was used in the kernel with 2 queues, the computation could not be performed as expected. But due to recent commits in the intel/sycl repository, the bug was fixed, so we need to add a verification test that reproduces this bug and verifies that the function works correctly

I'm not sure I understand how this test validates `depends_on` behavior.
According to my understanding if we remove `depends_on` call, the test will pass due to:
1. There is an explicit wait on `eA` before the second kernel starts.
2. Computations executed by the second kernel do not depend on the results produced by the first kernel.

What did I miss?